compiler/src/iree/compiler/GlobalOptimization/ExpandVectors.cpp

```diff
@@ -11,13 +11,45 @@
 
 #include "iree/compiler/GlobalOptimization/PassDetail.h"
 #include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Utils/Utils.h"
 #include "mlir/Dialect/Tensor/IR/Tensor.h"
 #include "mlir/Transforms/GreedyPatternRewriteDriver.h"
 
 namespace mlir {
 namespace iree_compiler {
 namespace GlobalOptimization {
 
+// If the producer is a CastOpInterface, or a linalg::GenericOp that performs
+// only a CastOpInterface on its input, return the CastOpInterface op
+static Operation *getDefiningCastOp(Value input) {
```

Comment:
Where should a util function like this live? It is somewhat specific to this matching, but it is needed in both ExpandVectors.cpp and SetEncoding.cpp now. Should we make a GlobalOptimization utils file?

Reply:
Yes, please create GlobalOptmization/Utils.[h|cpp] files and move the functions to there.